src/instructlab/model/serve_backend.py

```diff
@@ -23,6 +28,55 @@ def signal_handler(num_signal, __):
     sys.exit(0)
 
 
+def get_gpu_count() -> int:
+    # Third Party
+    import torch
+
+    if torch.cuda.is_available():
+        return torch.cuda.device_count()
+    return 0
+
+
+def check_gpu_count(gpus: int, available_gpus: int) -> bool:
+    return gpus > available_gpus
+
+
+def check_tensor_parallel_size(
+    vllm_args: List, available_gpus: int
+) -> Tuple[bool, int]:
+    """
+    Checks if --tensor-parallel-size argument is valid.
+
+    Returns:
+        - (True, tensor_parallel_size) if tensor-parallel-size is invalid (greater than available GPUs).
+        - (False, 0) if --tensor-parallel-size is not found or invalid.
+
+    Parameters:
+        vllm_args (List): Arguments from serve.vllm.vllm_args.
+
+    Example:
+        For vllm_args = ["--tensor-parallel-size", "4"], if there are 2 GPUs:
+            Returns: (True, 4)
+    """
+    if contains_argument("--tensor-parallel-size", vllm_args):
+        # First Party
+        from instructlab.model.backends.vllm import get_argument
+
+        tensor_parallel_size = 0
+        try:
+            tps = get_argument("--tensor-parallel-size", vllm_args)
+            if tps is not None:
+                tensor_parallel_size = int(tps)
+        except ValueError:
+            return (False, 0)
+        if tensor_parallel_size > available_gpus:
+            return (
+                True,
+                tensor_parallel_size,
+            )
+    return (False, 0)
```

Comment:
Is it desired to return 0 for tps when it's valid?  It might not break anything now but could get confused in the future.